include/envoy/config/BUILD

```diff
@@ -75,3 +75,23 @@ envoy_cc_library(
         "//source/common/protobuf",
     ],
 )
+
+envoy_cc_library(
+    name = "discovery_service_base_interface",
+    hdrs = ["discovery_service_base.h"],
+    deps = [
+        ":subscription_interface",
+        "//source/common/protobuf",
+        "@envoy_api//envoy/api/v2:pkg_cc_proto",
```

Comment:
Not a fan of pulling all xDS proto dependencies here. I guess if you move the `using` statements to each *Api classes this can be gone.